test/e2e/csv_e2e_test.go

```diff
@@ -4432,89 +4426,86 @@ var csvFailedChecker = buildCSVConditionChecker(operatorsv1alpha1.CSVPhaseFailed
 var csvAnyChecker = buildCSVConditionChecker(operatorsv1alpha1.CSVPhasePending, operatorsv1alpha1.CSVPhaseSucceeded, operatorsv1alpha1.CSVPhaseReplacing, operatorsv1alpha1.CSVPhaseDeleting, operatorsv1alpha1.CSVPhaseFailed)
 var csvCopiedChecker = buildCSVReasonChecker(operatorsv1alpha1.CSVReasonCopied)
 
-func fetchCSV(c versioned.Interface, name, namespace string, checker csvConditionChecker) (*operatorsv1alpha1.ClusterServiceVersion, error) {
-	var fetchedCSV *operatorsv1alpha1.ClusterServiceVersion
-	var err error
+func fetchCSV(c versioned.Interface, name, namespace string, checker csvConditionChecker) (fetched *operatorsv1alpha1.ClusterServiceVersion, err error) {
+	var lastMessage string
 
 	ctx.Ctx().Logf("waiting for CSV %s/%s to reach condition", namespace, name)
-	Eventually(func() (bool, error) {
-		fetchedCSV, err = c.OperatorsV1alpha1().ClusterServiceVersions(namespace).Get(context.TODO(), name, metav1.GetOptions{})
+	Eventually(func() bool {
```

Comment:
Can you please provide the reasoning behind your changes? I've stared at the `awaitCSV` -> `fetchCSV` change for a bit and I'm, not really sure what the difference is.

Reply:
Did you remove one of them, or is there a change to the code within?

Reply:
Yes to both. `awaitCSV` is gone. `fetchCSV` is more robust.

Reply:
And this was basically a loop around a function that loops.